third_party/move/move-compiler-v2/tests/file-format-generator/pack_unpack_ref.exp

```diff
@@ -47,10 +49,11 @@ B0:
 	12: Ret
 }
 unpack_ref_G(Arg0: &G): u64 * u64 * u64 * u64 /* def_idx: 2 */ {
-L0:	loc1: &u64
-L1:	loc2: &u64
-L2:	loc3: &u64
-L3:	loc4: &u64
+L0:	loc0: &S
```

Comment:
Look down below at the first use of `loc0`.  It is on the statement`StLoc[1](loc0: &S)`, where the `1` indicates parameter at index 1.  In that case, I guess it's `parameter_or_local`, not `local`.  It may be more useful to label these in the list as `L1:    loc0: &S`

Reply:
> Look down below at the first use of loc0. It is on the statementStLoc[1](loc0: &S), where the 1 indicates parameter at index 1.

No, `1` cannot indicate the parameter at index 1, because there is only 1 parameter (`Arg0`) whose index is 0.

`1` here indicates a slot into function locals, which I believe is laid out with all the params first, followed by all the non-param locals.

Reply:
In `disassembler.rs`, we have 2 indexes for each local: as a `local`, and as a `parameter_or_local`.  The list at the start of the function is list just containing locals, so after `.enum()`, the index is as a `local`.  In the instruction list, the index shown (e.g., `[1]` in `StLocal[1](loc0: &S)`) is as a `parameter_or_local`.  To show have the numbers agree between `L1` and `[1]`, you should use `self.type_for_local_or_parameter(parameter_len + local_idx, ...` in  `disassembler.rs`.

Reply:
parameter `Arg0` has index `0`.  By adding `parameter_len` to the `local` index you avoid conflicting with it.

Reply:
Ok, I think there were a couple of points that confused me about what you were asking for:
1. `1` indicates parameter at index `1` in `StLoc[1]` (which is not true, it is an index into what you later call `parameter_or_local` or what the move-binary-format bytecode refers to as "function locals" in its documentation).
2. The fix you suggest below and repeat here causes out of index error (because this is not where the `L{num}` label is being computed, see my latest commit for this).

I now see what you asking for is simply an additional feature request (rather than a bug fix, which we originally started out to fix here, where we would refer to non-existent items) to say, local labels, instead of starting from `L0`, should instead align with indexes used in the instructions. I agree with you that it is reasonable to do this change, which is IMO better than what we had. The latest commit now has this.